From f80d8870b59161561f92eb7ec406bc95dcf2dd3c Mon Sep 17 00:00:00 2001
From: Alexandru Gagniuc <mr.nuke.me@gmail.com>
Date: Tue, 6 Oct 2020 14:07:12 -0500
Subject: [PATCH] [HACK] plat-stm32mp1: tzc400: Replace panic with error
 message

The first stage bootloader is responsible for configuring the TZDRAM
access regions. This does not yet happen, so disable the checks for
the time being.
---
 core/arch/arm/plat-stm32mp1/plat_tzc400.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/core/arch/arm/plat-stm32mp1/plat_tzc400.c b/core/arch/arm/plat-stm32mp1/plat_tzc400.c
index bce92371..fad3fc12 100644
--- a/core/arch/arm/plat-stm32mp1/plat_tzc400.c
+++ b/core/arch/arm/plat-stm32mp1/plat_tzc400.c
@@ -85,20 +85,20 @@ static TEE_Result init_stm32mp1_tzc(void)
 	if (dram_start < tzdram_start) {
 		if (!tzc_region_is_non_secure(region_index, dram_start,
 					      tzdram_start - dram_start))
-			panic("Unexpected TZC area on non-secure region");
+			EMSG("Unexpected TZC area on non-secure region");
 
 		region_index++;
 	}
 
 	if (!tzc_region_is_secure(region_index, tzdram_start, tzdram_size))
-		panic("Unexpected TZC configuration on secure region");
+		EMSG("Unexpected TZC configuration on secure region");
 
 	if (tzdram_end < dram_end) {
 		region_index++;
 
 		if (!tzc_region_is_non_secure(region_index, tzdram_end,
 					      dram_end - tzdram_end))
-			panic("Unexpected TZC area on non-secure region");
+			EMSG("Unexpected TZC area on non-secure region");
 	}
 
 	itr_add(&tzc_itr_handler);
-- 
2.26.2

